oak/server/channel.h

```diff
@@ -102,12 +103,24 @@ class MessageChannel {
 
   mutable absl::Mutex mu_;  // protects msgs_
   std::deque<std::unique_ptr<Message>> msgs_ GUARDED_BY(mu_);
+
+  mutable absl::Mutex count_mu_;  // protects read_count_, write_count_
```

Comment:
How about using `atomic` ints?

Reply:
I try to avoid using raw `atomic`s where possible, e.g. [advice](https://abseil.io/docs/cpp/guides/synchronization#memory-access-issues): "Note that the rules for properly applying atomic operations are quite complicated, which is one of many reasons you should avoid atomics."